<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>706</width>
    <height>423</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>labscript runmanager</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
    <normaloff>:/custom/runmanager.png</normaloff>:/custom/runmanager.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">QPushButton {
     border: none;
     padding: 4px;
 }

QPushButton:hover {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #f6f7fa, stop: 1 #dadbde);
     border: 1px solid #8f8f91;
     border-radius: 3px;
 }

 QPushButton:pressed {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #dadbde, stop: 1 #f6f7fa);
 }

QPushButton:checked {
     background-color: #dadbde;
     border: 1px solid #8f8f91;
     border-radius: 3px;
 }

QPushButton:hover:checked {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #dadbde, stop: 1 #f6f7fa);
     border: 1px solid #8f8f91;
     border-radius: 3px;
 }

QToolButton {
     border: none;
 }

QToolButton:hover {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #f6f7fa, stop: 1 #dadbde);
     border: 1px solid #8f8f91;
     border-radius: 3px;
 }

 QToolButton:pressed {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #dadbde, stop: 1 #f6f7fa);
 }

QToolButton:checked {
     background-color: #dadbde;
     border: 1px solid #8f8f91;
     border-radius: 3px;
 }

QToolButton:hover:checked {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #dadbde, stop: 1 #f6f7fa);
     border: 1px solid #8f8f91;
     border-radius: 3px;
 }
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="spacing">
       <number>0</number>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QPushButton" name="pushButton_engage">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Compile pending shots and submit them to BLACS if applicable.&lt;/p&gt;&lt;p&gt;Or, send the parameter space description to mise, if that option is selected.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="autoFillBackground">
            <bool>false</bool>
           </property>
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="text">
            <string>Engage</string>
           </property>
           <property name="icon">
            <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
             <normaloff>:/fugue/control.png</normaloff>:/fugue/control.png</iconset>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
           <property name="flat">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="pushButton_abort">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Stop compiling shots after the current shot is compiled.
 To interrupt a compilation mid-shot, use the 'Restart subprocess' button</string>
           </property>
           <property name="text">
            <string>Abort</string>
           </property>
           <property name="icon">
            <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
             <normaloff>:/fugue/cross-octagon.png</normaloff>:/fugue/cross-octagon.png</iconset>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="pushButton_restart_subprocess">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Forcefully restart the compilation subprocess, stopping any compilation in process.&lt;/p&gt;&lt;p&gt;This can be useful if the child process is misbehaving for any reason.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Restart
subprocess</string>
           </property>
           <property name="icon">
            <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
             <normaloff>:/fugue/arrow-circle-135-left.png</normaloff>:/fugue/arrow-circle-135-left.png</iconset>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="pushButton_shuffle">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Send compiled shots to BLACS in a random order, to prevent scanned parameters correlating with temporal drifts.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Shuffle</string>
           </property>
           <property name="icon">
            <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
             <normaloff>:/fugue/arrow-switch.png</normaloff>:/fugue/arrow-switch.png</iconset>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="spacing">
          <number>9</number>
         </property>
         <property name="sizeConstraint">
          <enum>QLayout::SetFixedSize</enum>
         </property>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QRadioButton" name="radioButton_compile">
             <property name="toolTip">
              <string>Compile shots for the currently described parameter space</string>
             </property>
             <property name="text">
              <string>Compile</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_send_to_mise">
             <property name="toolTip">
              <string>Send a description of the parameter space to mise, which will compile shots itself.</string>
             </property>
             <property name="text">
              <string>Send to mise</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QStackedWidget" name="stackedWidget">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="stackedWidgetPage_compile">
            <layout class="QFormLayout" name="formLayout">
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
             <item row="0" column="0">
              <widget class="QCheckBox" name="checkBox_run_shots">
               <property name="toolTip">
                <string>Send the shot files to BLACS after compilation</string>
               </property>
               <property name="text">
                <string>Run shot(s)</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QCheckBox" name="checkBox_view_runs">
               <property name="toolTip">
                <string>Open the shot files in runviewer after compilation</string>
               </property>
               <property name="text">
                <string>View run(s)</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="stackedWidgetPage_send_to_mise">
            <layout class="QFormLayout" name="formLayout_2">
             <property name="margin">
              <number>0</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>mise hostname</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="lineEdit_mise_hostname">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>The host computer running mise</string>
               </property>
               <property name="text">
                <string>localhost</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <layout class="QGridLayout" name="gridLayout_3">
         <property name="topMargin">
          <number>6</number>
         </property>
         <property name="bottomMargin">
          <number>6</number>
         </property>
         <item row="0" column="1">
          <widget class="QGroupBox" name="groupBox">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLineEdit" name="lineEdit_labscript_file">
              <property name="toolTip">
               <string>The labscript file to compile</string>
              </property>
              <property name="frame">
               <bool>true</bool>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="toolButton_select_labscript_file">
              <property name="toolTip">
               <string>Select a file ...</string>
              </property>
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
                <normaloff>:/custom/python-document.png</normaloff>:/custom/python-document.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QGroupBox" name="groupBox">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLineEdit" name="lineEdit_shot_output_folder">
              <property name="toolTip">
               <string>The folder in which newly compiled shot files will be placed</string>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="toolButton_select_shot_output_folder">
              <property name="toolTip">
               <string>Select folder ...</string>
              </property>
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
                <normaloff>:/fugue/folder-horizontal-open.png</normaloff>:/fugue/folder-horizontal-open.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>labscript file</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QToolButton" name="toolButton_edit_labscript_file">
           <property name="toolTip">
            <string>Edit this labscript file in a text editor</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
             <normaloff>:/fugue/document--pencil.png</normaloff>:/fugue/document--pencil.png</iconset>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Shot output folder</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QToolButton" name="toolButton_reset_shot_output_folder">
           <property name="toolTip">
            <string>Reset to default output folder from labconfig</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
             <normaloff>:/fugue/arrow-turn-180-left.png</normaloff>:/fugue/arrow-turn-180-left.png</iconset>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>BLACS hostname</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="lineEdit">
           <property name="toolTip">
            <string>The host computer running BLACS</string>
           </property>
           <property name="text">
            <string>localhost</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="FingerTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="tabPosition">
       <enum>QTabWidget::West</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>2</number>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideNone</enum>
      </property>
      <property name="usesScrollButtons">
       <bool>false</bool>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tab_output">
       <attribute name="icon">
        <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
         <normaloff>:/fugue/terminal.png</normaloff>:/fugue/terminal.png</iconset>
       </attribute>
       <attribute name="title">
        <string>Output</string>
       </attribute>
       <attribute name="toolTip">
        <string>View textual output of compilation</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_output_tab"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_overview">
       <attribute name="icon">
        <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
         <normaloff>:/custom/outer.png</normaloff>:/custom/outer.png</iconset>
       </attribute>
       <attribute name="title">
        <string>Overview</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="spacing">
           <number>9</number>
          </property>
          <property name="leftMargin">
           <number>3</number>
          </property>
          <property name="rightMargin">
           <number>3</number>
          </property>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="toolButton_axis_to_top">
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
              <normaloff>:/fugue/arrow-stop-090.png</normaloff>:/fugue/arrow-stop-090.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="toolButton_axis_up">
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
              <normaloff>:/fugue/arrow-090.png</normaloff>:/fugue/arrow-090.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="toolButton_axis_down">
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
              <normaloff>:/fugue/arrow-270.png</normaloff>:/fugue/arrow-270.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="toolButton_axis_to_bottom">
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
              <normaloff>:/fugue/arrow-stop-270.png</normaloff>:/fugue/arrow-stop-270.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTreeView" name="treeView_overview"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_groups">
       <attribute name="icon">
        <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
         <normaloff>:/fugue/tables-stacks.png</normaloff>:/fugue/tables-stacks.png</iconset>
       </attribute>
       <attribute name="title">
        <string>Groups</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <property name="spacing">
           <number>9</number>
          </property>
          <property name="margin">
           <number>3</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton_open_globals_file">
            <property name="toolTip">
             <string>Load an HDF file containing globals into runviewer</string>
            </property>
            <property name="text">
             <string>Open globals file</string>
            </property>
            <property name="icon">
             <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
              <normaloff>:/fugue/folder-open-table.png</normaloff>:/fugue/folder-open-table.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_new_globals_file">
            <property name="toolTip">
             <string>Create a new HDF file to store globals in</string>
            </property>
            <property name="text">
             <string>New globals file</string>
            </property>
            <property name="icon">
             <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
              <normaloff>:/fugue/table--plus.png</normaloff>:/fugue/table--plus.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_diff_globals_file">
            <property name="toolTip">
             <string>Compare the globals stored in an HDF file with those currently in use by runviewer</string>
            </property>
            <property name="text">
             <string>Diff globals file</string>
            </property>
            <property name="icon">
             <iconset resource="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc">
              <normaloff>:/fugue/tables.png</normaloff>:/fugue/tables.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTreeView" name="treeView_groups"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>706</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_configuration"/>
    <addaction name="actionSave_configuration"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionLoad_configuration">
   <property name="text">
    <string>Load configuration</string>
   </property>
  </action>
  <action name="actionSave_configuration">
   <property name="text">
    <string>Save configuration</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>FingerTabWidget</class>
   <extends>QTabWidget</extends>
   <header>qtutils/widgets.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../Anaconda/Lib/site-packages/qtutils-1.3.0-py2.7.egg/qtutils/icons/icons.qrc"/>
 </resources>
 <connections/>
</ui>
